java/maven/test/unit/src/org/netbeans/modules/maven/queries/UnitTestsCompilerOptionsQueryImplTest.java

```diff
@@ -59,13 +60,19 @@ public void testNoCompilerPluginSpecified() throws Exception {
         TestFileUtils.writeFile(wd,
                                 "src/test/java/module-info.java",
                                 "module test { requires testng; }\n");
+        TestFileUtils.writeFile(wd,
+                                "target/generated-sources/java/test/Gen.java",
+                                "package test;\n" +
+                                "public class Gen {}\n");
         FileObject testSource =
         TestFileUtils.writeFile(wd,
                                 "src/test/java/test/APITest.java",
                                 "package test;\n" +
                                 "public class APITest {}\n");
         assertEquals(Arrays.asList("--patch-module",
-                                   "test=" + FileUtil.toFile(wd.getFileObject("src/main/java")).getAbsolutePath()),
+                                   "test=" + FileUtil.toFile(wd.getFileObject("src/main/java")).toURI().getPath() +
```

Comment:
Are paths or URIs expected for patch-module? If paths are expected, this will break for example for spaces ("%20" vs " "). The documentation seems to be thin on this.

Reply:
Paths are expected. It is a good comment on escaping, but isn't URI.getPath() returning un-escaped paths? (I've tweaked the test to actually use a space with paths.)

Reply:
You are right, I had to reiterate the difference between java.net.URL and java.net.URI and I had to realize that an URI is used here.